src/socket.rs

```diff
@@ -861,6 +861,41 @@ impl Socket {
         }
     }
 
+    /// IP_TRANSPARENT (since Linux 2.6.24)
+    /// Setting this boolean option enables transparent proxying
+    /// on this socket.  This socket option allows the calling
+    /// application to bind to a nonlocal IP address and operate
+    /// both as a client and a server with the foreign address as
+    /// the local endpoint.  NOTE: this requires that routing be
+    /// set up in a way that packets going to the foreign address
+    /// are routed through the TProxy box (i.e., the system
+    /// hosting the application that employs the IP_TRANSPARENT
+    /// socket option).  Enabling this socket option requires
+    /// superuser privileges (the CAP_NET_ADMIN capability).
+    ///
+    /// TProxy redirection with the iptables TPROXY target also
+    /// requires that this option be set on the redirected socket.
+    /// this feature is only available on linux
+    #[cfg(any(target_os = "linux"))]
+    pub fn set_ip_transparent(&self, transparent: bool) -> io::Result<()> {
+        unsafe {
+            setsockopt(
+                self.inner,
+                sys::IPPROTO_IP,
+                libc::IP_TRANSPARENT,
+                transparent as c_int,
+            )
+        }
+    }
+    /// Get whether the IP_TRANSPARENT is set for this socket.
```

Comment:
Could you following the convention of the other settings. I.e. Get the value of the `IP_TRANSPARENT` option on this socket.